app/controllers/AMIable.scala

```diff
@@ -28,11 +28,12 @@ class AMIable (val controllerComponents: ControllerComponents, val amiableConfig
     attempt {
       for {
         instancesWithAmis <- Prism.instancesWithAmis(ssa)
+        accountNames = PrismLogic.instanceSSAAs(instancesWithAmis.map(_._1)).flatMap(_.accountName).distinct
```

Comment:
This logic is probably complex enough to be pulled out into its own function and tested. I try to avoid putting any "logic" into an application's "edge for comprehensions". The threshold for "logic" is a super fuzzy one though. As a rule of thumb, I allow myself to pass function names to map and traverse but any actual implementation beyond this is where my logic spidey-senses start to kick in.

In this case the swinger might be that this ought really to be tested. We have a few important properties here that we probably want to enforce to maintain our contract with the template function:
* there are no duplicates
* every instance will find its corresponding account here
* we are taking the account name rather than some other string <- this one is only borderline worth the test code :-)

Reply:
> Is there any reason why this needs to be drawn specifically from the instance accounts while the ssaInstanceAMIs fn below can use all the accounts from Prism?

No, except to save an extra HTTP call as we already have that data available here. Maybe for consistency I should just call out to prism in both cases.

Re pulling logic out of the controller - yep that makes sense.

Reply:
Update - I switched to using the same endpoint as `ssaInstanceAMIs`, which simultaneously reduces the logic in the controller down to a single map operation